go.mod

```diff
@@ -3,18 +3,18 @@ module github.com/capi-samples/cluster-api-provider-kwok
 go 1.19
 
 require (
-	github.com/go-logr/logr v1.2.3
-	github.com/onsi/ginkgo/v2 v2.9.2
-	github.com/onsi/gomega v1.27.5
+	github.com/go-logr/logr v1.2.4
+	github.com/onsi/ginkgo/v2 v2.11.0
+	github.com/onsi/gomega v1.27.8
 	github.com/pkg/errors v0.9.1
 	github.com/spf13/pflag v1.0.5
-	k8s.io/api v0.27.1
-	k8s.io/apimachinery v0.27.1
-	k8s.io/client-go v0.27.1
+	k8s.io/api v0.27.2
+	k8s.io/apimachinery v0.27.2
+	k8s.io/client-go v0.27.2
 	k8s.io/klog/v2 v2.90.1
 	k8s.io/utils v0.0.0-20230209194617-a36077c30491
-	sigs.k8s.io/cluster-api v1.4.1
-	sigs.k8s.io/controller-runtime v0.14.5
```

Comment:
This would throw the following error
![image](https://github.com/capi-samples/cluster-api-provider-kwok/assets/34534103/1ba75164-f0a7-4940-ae1c-ef692ed9a265)

Updating controller-runtime to `v0.15.0` fixes the above issue but creates another problem. There's a bunch of [breaking changes in `v0.15.0`](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.15.0) which cluster-api has addressed in `v1.5.0-beta.0`. Any lower version of cluster-api and `go build` fails.